cli/worker.rs

```diff
@@ -428,6 +429,7 @@ impl CliMainWorkerFactory {
     permissions: PermissionsContainer,
     mut custom_extensions: Vec<Extension>,
     stdio: deno_runtime::deno_io::Stdio,
+    is_test_or_jupyter: bool,
```

Comment:
😢 do we need to explicitly pass a boolean here? Maybe pass a `DenoSubcommand` enum here and do a match instead?

Reply:
Sure but its a bit frustrating how much complicated this cli factory bit is, too much option passing going on.